jetbrains-core/src/software/aws/toolkits/jetbrains/services/s3/objectActions/CopyPathAction.kt

```diff
@@ -3,22 +3,21 @@
 
 package software.aws.toolkits.jetbrains.services.s3.objectActions
 
+import com.intellij.openapi.actionSystem.CommonDataKeys
+import com.intellij.openapi.actionSystem.DataContext
 import com.intellij.openapi.ide.CopyPasteManager
-import com.intellij.openapi.project.Project
-import software.aws.toolkits.jetbrains.services.s3.editor.S3TreeContinuationNode
-import software.aws.toolkits.jetbrains.services.s3.editor.S3TreeErrorNode
+import software.aws.toolkits.jetbrains.core.utils.getRequiredData
 import software.aws.toolkits.jetbrains.services.s3.editor.S3TreeNode
 import software.aws.toolkits.jetbrains.services.s3.editor.S3TreeObjectVersionNode
-import software.aws.toolkits.jetbrains.services.s3.editor.S3TreeTable
 import software.aws.toolkits.resources.message
 import software.aws.toolkits.telemetry.S3Telemetry
 import java.awt.datatransfer.StringSelection
 
-class CopyPathAction(private val project: Project, treeTable: S3TreeTable) : SingleS3ObjectAction(treeTable, message("s3.copy.path")) {
-    override fun performAction(node: S3TreeNode) {
+class CopyPathAction : SingleS3ObjectAction(message("s3.copy.path")) {
+    override fun performAction(dataContext: DataContext, node: S3TreeNode) {
         CopyPasteManager.getInstance().setContents(StringSelection(node.key))
-        S3Telemetry.copyPath(project)
+        S3Telemetry.copyPath(dataContext.getRequiredData(CommonDataKeys.PROJECT))
     }
 
-    override fun enabled(node: S3TreeNode): Boolean = node !is S3TreeObjectVersionNode && node !is S3TreeContinuationNode<*> && node !is S3TreeErrorNode
```

Comment:
why was `&& node !is S3TreeContinuationNode<*> && node !is S3TreeErrorNode` removed?

Reply:
DataContext filters out those nodes

https://github.com/aws/aws-toolkit-jetbrains/pull/2401/files#diff-b26b311a1cfe610f3d4b3dea94c8e83935031bd9dce7a6479963c123e0f98637R65

Reply:
But maybe we shouldn't? So that selecting one of those and a valid node will disable these actions?

Reply:
can we have tests on that data provider?

Reply:
Ya, we should get those added

Reply:
> But maybe we shouldn't? So that selecting one of those and a valid node will disable these actions?

how about just making errors/continuation nodes nonselectable?

Reply:
I went looking for a Swing API for that and did not find one. Do you know of a way?